airflow_prometheus_exporter/prometheus_exporter.py

```diff
@@ -149,6 +155,44 @@ def get_task_failure_counts():
             TaskFail.task_id,
         )
 
+def get_xcom_params(task_id):
+    #get latest task_id, join to xcom table
+    with session_scope(Session) as session:
+        max_execution_dt_query = session.query(
+            DagRun.dag_id,
+            func.max(DagRun.execution_date).label('max_execution_dt')
+        ).group_by(
+            DagRun.dag_id
+        ).subquery()
+
+        return session.query(
+            XCom.dag_id,
+            XCom.value
+        ).join(max_execution_dt_query,
+               and_(
+                   (
+                    XCom.dag_id == max_execution_dt_query.c.dag_id
+                   ),
+                   (XCom.execution_date == max_execution_dt_query.c.max_execution_dt
+                   )
+               )).filter(XCom.task_id == task_id).all()
+
+
+def extract_xcom_parameter(value):
+    enable_pickling = conf.getboolean('core', 'enable_xcom_pickling')
+    if enable_pickling:
+        return pickle.loads(value)
+    else:
+        try:
+            return json.loads(value.decode('UTF-8'))
+        except ValueError:
+            log = LoggingMixin().log
+            log.error("Could not deserialize the XCOM value from JSON. "
+                      "If you are using pickles instead of JSON "
+                      "for XCOM, then you need to enable pickle "
+                      "support for XCOM in your airflow config.")
+            raise
```

Comment:
What would happen if this raises an exception? Would this cause the exporter to fail?

Reply:
yes...good point...what if i just return an empty dict?